File: poo/3. Newton's Laws Simulator/forces.php
<?php
    abstract class forces {
        readonly private ?string $name;
        readonly private ?string $nameSpanish;

        public function __construct(?string $name, ?string $nameSpanish) {
            $this->name = $name;
            $this->nameSpanish = $nameSpanish;
        }
        public function getName(): ?string {
            return $this->name;
        }
        public function getNameSpanish(): ?string {
            return $this->nameSpanish;
        }

        abstract public function calculateForce(): ?float;
    }
?>

File: poo/3. Newton's Laws Simulator  JS/forces.java
public abstract class Forces {
    private final String name;
    private final String nameSpanish;

    public Forces(String name, String nameSpanish) {
        this.name = name;
        this.nameSpanish = nameSpanish;
    }

    public String getName() {
        return name;
    }

    public String getNameSpanish() {
        return nameSpanish;
    }

    public abstract Float calculateForce();
}
